spec/index.bs

```diff
@@ -4090,10 +4052,57 @@ enum GPUCullMode {
 };
 </script>
 
-### Color State ### {#color-state}
+### Multisample State ### {#multisample-state}
+
+<script type=idl>
+dictionary GPUMultisampleState {
+    GPUSize32 count = 1;
+    GPUSampleMask mask = 0xFFFFFFFF;
+    boolean alphaToCoverageEnabled = false;
+};
+</script>
+
+<div algorithm>
+    <dfn abstract-op>validating GPUMultisampleState</dfn>(|descriptor|)
+        **Arguments:**
+            - {{GPUMultisampleState}} |descriptor|
+
+        Return `true` if all of the following conditions are satisfied:
+            - If |descriptor|.{{GPUMultisampleState/alphaToCoverageEnabled}} is `true`:
+                - |descriptor|.{{GPUMultisampleState/count}} is greater than 1.
+</div>
+
+### Fragment State ### {#fragment-state}
 
 <script type=idl>
-dictionary GPUColorStateDescriptor {
+dictionary GPUFragmentState: GPUProgrammableStage {
+    required sequence<GPUColorTargetState> outputTargets;
+};
+</script>
+
+<div algorithm>
+    <dfn abstract-op>validating GPUFragmentState</dfn>(|descriptor|)
+        Return `true` if all of the following conditions are satisfied:
+
+            - |descriptor|.{{GPUFragmentState/outputTargets}}.length is less than or equal to 4.
+            - For each |colorState| layout descriptor in the list |descriptor|.{{GPUFragmentState/outputTargets}}:
+                - |colorState|.{{GPUColorTargetState/format}} is listed in {#plain-color-formats}
+                    with {{GPUTextureUsage/RENDER_ATTACHMENT}} capability.
+                - |colorState|.{{GPUColorTargetState/blend}} is either `undefined`,
+                    or the |colorState|.{{GPUColorTargetState/format}} is filterable
+                    according to the {#plain-color-formats} table.
+                - |colorState|.{{GPUColorTargetState/writeMask}} is less than 16.
+                - |descriptor|.{{GPUProgrammableStage/module}} contains an output variable
+                    that is [=statically used=] by |descriptor|.{{GPUProgrammableStage/entryPoint}},
+                    and has a type that is compatible with |colorState|.{{GPUColorTargetState/format}}.
+</div>
+
+Issue: define the area of reach for "statically used" things of `GPUProgrammableStage`
+
+### Color Target State ### {#color-target-state}
+
+<script type=idl>
+dictionary GPUColorTargetState {
     required GPUTextureFormat format;
 
     GPUBlendDescriptor blend;
```

Comment:
Silly non-blocking nit: Given some of the other changes that are trying to reduce the usage of "Descriptor" this seems like it should be `GPUBlendState`

Reply:
right, right, I'll do it now!